nameserver/nameserver.go

```diff
@@ -221,15 +221,20 @@ func (n *Nameserver) receiveGossip(msg []byte) (mesh.GossipData, mesh.GossipData
 					nextVersion := e.Version + 1
 					*e = *ourEntry
 					e.Version = nextVersion
+					entriesToBroadcast = append(entriesToBroadcast, *e)
 				}
 			} else { // We have no entry matching the one that came in with us as Origin
 				e.tombstone()
+				entriesToBroadcast = append(entriesToBroadcast, *e)
 			}
 		}
 		return true
 	})
 
 	newEntries := n.entries.merge(gossip.Entries)
+	n.Unlock() // unlock before attempting to broadcast
+	n.broadcastEntries(entriesToBroadcast...)
```

Comment:
There is a subtlety here... the entries we are broadcasting are (potentially updated) entries from the incoming gossip, rather than entries we hold. It just so happens that they also happen to correspond to identical entries in the latter, post merge. Which is rather important. That's the subtlety.

Might deserve a comment.